openapi/api.yml

```diff
@@ -149,6 +149,26 @@ paths:
                 $ref: '#/components/schemas/account'
         404:
           description: "No associated accounts found."
+  /accounts/search:
+    post:
+      tags:
+        - accounts
+      summary: Search for accounts based on certain criteria
+      operationId: core.account.search_accounts
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/accountSearch'
+      responses:
```

Comment:
Related to the other comment about required fields, do we ever need to account for the endpoint returning no results? If no args are passed and no filters applied we'd get all the accounts back, but there's potentially a combo of filters (an email domain and a role) that could return nothing if there's no one with that role and email domain? And also some of the tests are checking for things that return 0 results, just wondering if that should still be a 200 or handled differently.

Reply:
It is valid for a search to return no results - the answer to the question is just an empty list. It doesn't make sense to return a 404 in that case - that's more about saying the resource doesn't exist. In this case the resource is a search (result) - which you will always get.

In my opinion, `get_accounts_for_fund` is doing the wrong thing by returning a 404 for their search 🙃 🤐

Reply:
Fair point, thanks for clarifying!